### PR TITLE
http client accepting invalid hostnames

### PR DESCRIPTION
**Pull request by varas**
#### Description of the changes

Package `http` can now create a client that validate certificates but also  accepts invalid hostnames via `NewAcceptInvalidHostname`.

This addresses issue: https://github.com/newrelic/infra-integrations-sdk/issues/185

### Author

- [x] add a risk label after carefully considering the "blast radius" of your changes
- [x] describe the _intent_ of your changes in the description. don't just rewrite your code in prose
- [x] assign at least one reviewer
- [x] document feature

### Reviewer

- [ ] review code for readability
- [ ] verify that high risk behavior changes are well tested
- [ ] check license for any new external dependency
- [ ] request documenation about anything that isn't clear and obvious
- [ ] document agent version supporting the feature

### CODE REVIEW

**Comment by varas** [comment]
go 1.8 incompatibility on helper :sigh:

File: http/client.go [hunk 1]
```diff
@@ -16,37 +17,89 @@ import (
...
+	return _new(CABundleFile, CABundleDir, httpTimeout, "")
+}
+
+// NewAcceptInvalidHostname new http.Client with ability to accept HTTPS certificates that don't
+// match the hostname of the server they are connecting to.
+func NewAcceptInvalidHostname(CABundleFile, CABundleDir string, httpTimeout time.Duration, hostname string) (*http.Client, error) {
+	return _new(CABundleFile, CABundleDir, httpTimeout, hostname)
+}
+
+func _new(CABundleFile, CABundleDir string, httpTimeout time.Duration, acceptInvalidHostname string) (*http.Client, error) {
```
Comment (ardias) [inline http/client.go:29]: just a small comment: maybe **acceptInvalidHostname** would be better named something like "alternateHostname"

Comment (varas) [reply http/client.go:29]: I rather the prior as it explicitly says it comes from its caller `NewAcceptInvalidHostname` in comparison to its `New` counterpart

Comment (varas) [reply http/client.go:29]: but if you have a strong opinion be my guest

**Pull request merged**